pkgs/applications/science/logic/uhdm/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, python3
+}:
+
+stdenv.mkDerivation rec {
+  pname = "UHDM";
+  version = "0.9.1.40";
+
+  src = fetchFromGitHub {
+    owner = "chipsalliance";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-CliKU2WM8B9012aDcS/mTyIf+JcsVsc4uRRi9+FRWbM=";
+    fetchSubmodules = true;
+  };
+
```

Comment:
There is something strange [going on in darwin](https://github.com/NixOS/nixpkgs/pull/173225/checks?check_run_id=8585878593) where it fails to compile as capnp does not seem to find the biuld prefix and then does not produce the generated file. Maybe something different going on on that platform with symbolic links ?

I suggest that we attempt to add this and see if that makes compilation on Darwin happy.

```suggestion
  prePatch = ''                                                                                                                                                                                               
     substituteInPlace CMakeLists.txt --replace \                                                                                                                                                             
       'capnp compile' \                                                                                                                                                                                      
       'capnp compile --src-prefix=''${GENDIR}/..'                                                                                                                                                            
  '';

```